lib/src/map_recorder.dart

```diff
@@ -0,0 +1,106 @@
+part of mapbox_maps_flutter;
+
+/// MapRecorder provides functions to record and replay API calls of a [MapboxMap] instance.
+///
+/// These recordings can be used to debug issues which require multiple steps to reproduce.
+/// Additionally, playbacks can be used for performance testing custom scenarios.
+///
+/// **Note**: The raw format produced by [stopRecording] is experimental and there is no guarantee
+/// for version cross-compatibility when feeding it to [replay].
+///
+/// **Warning**: This API is experimental and may change in future releases.
+///
+/// **Example**:
+/// ```dart
+/// // Start recording
+/// await mapboxMap.recorder.startRecording(
+///   timeWindow: 60000, // 60 seconds
+///   loggingEnabled: true,
+///   compressed: true,
+/// );
+///
+/// // ... perform some map interactions ...
+///
+/// // Stop recording and get the recorded sequence
+/// final recordedSequence = await mapboxMap.recorder.stopRecording();
+///
+/// // Replay the sequence twice at 2x speed
+/// await mapboxMap.recorder.replay(
+///   recordedSequence,
+///   playbackCount: 2,
+///   playbackSpeedMultiplier: 2.0,
+///   avoidPlaybackPauses: false,
+/// );
+/// ```
+@experimental
+class MapRecorder {
+  final _MapRecorderMessenger _messenger;
+
+  MapRecorder._(this._messenger);
+
+  /// Begins the recording session.
+  ///
+  /// [timeWindow] - The maximum duration (in milliseconds) from the current time until API calls are kept.
+  /// If not specified, all API calls will be kept during the recording, which can lead to significant memory consumption for long sessions.
+  ///
+  /// [loggingEnabled] - If set to true, the recorded API calls will be printed in the logs. Default value: false.
+  ///
+  /// [compressed] - If set to true, the recorded output will be compressed with gzip. Default value: false.
+  Future<void> startRecording({
+    int? timeWindow,
```

Comment:
Should we use Duration here instead? Should be trivial to convert to int inside the method